src/blockchain_utilities/blockchain_import.cpp

```diff
@@ -487,10 +487,19 @@ int import_from_file(cryptonote::core& core, const std::string& import_file_path
           cumulative_difficulty = bp.cumulative_difficulty;
           coins_generated = bp.coins_generated;
 
+          std::unordered_map<uint64_t, rct::key> transparent_amount_commitments;
+          {
+            std::vector<std::reference_wrapper<const transaction>> tx_refs{std::cref(b.miner_tx)};
+            tx_refs.reserve(1 + txs.size());
```

Comment:
Very nit-picky, but you should always call `reserve()` *before* you add the first element (`std::cref(b.miner_tx)` in this case). Because as soon as you add the first element, the `std::vector` *has* to allocate, an the call to `reserve` is too late. IIRC, in g++'s stdlib, the vector is *unallocated* after the default constructor, which means the call to `reserve()` triggers the first allocation.